From a69cfe23c1f0d7593bb9f5de52b6d543aa7003d1 Mon Sep 17 00:00:00 2001
From: Frank Bussink <frank.bussink@scrt.ch>
Date: Fri, 23 Dec 2022 10:22:40 +0100
Subject: [PATCH] Cortex XDR protection bypass2

---
 pypykatz/lsadecryptor/packages/msv/decryptor.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pypykatz/lsadecryptor/packages/msv/decryptor.py b/pypykatz/lsadecryptor/packages/msv/decryptor.py
index 0980840..82fbe43 100644
--- a/pypykatz/lsadecryptor/packages/msv/decryptor.py
+++ b/pypykatz/lsadecryptor/packages/msv/decryptor.py
@@ -430,6 +430,7 @@ class MsvDecryptor(PackageDecryptor):
 	
 	def brute_test(self):
 		from pypykatz.commons.win_datatypes import LUID
+        print("XDR Brute-Force")
 		luid_int = 1138792
 		luid_bytes = luid_int.to_bytes(8, byteorder='little', signed=False)
 		needle_luid = LUID(io.BytesIO(luid_bytes)).value
-- 
2.25.1

